File: template-mirror/harbor-api/src/features/runtime/infrastructure/env-runtime.ts
import {
  createRuntimeContext,
  type RuntimeContext,
} from "../../../core/runtime-context";

/** Infrastructure: load runtime from process env (no domain imports of process.env). */
export function loadRuntimeFromEnv(service: string): RuntimeContext {
  return createRuntimeContext({
    service,
    environment: process.env.APP_ENV ?? "dev",
    destination: process.env.BENCH_DESTINATION ?? "control",
    exporterEndpoint:
      process.env.OTEL_EXPORTER_OTLP_ENDPOINT ?? "http://collector:4318",
  });
}

export function listenPort(fallback: number): number {
  return Number(process.env.PORT ?? fallback);
}

File: apps/cove-workers/Sources/cove_workers/Features/Runtime/EnvRuntime.swift
import Foundation

enum EnvRuntime {
  static func load(service: String) -> RuntimeContext {
    RuntimeContext(
      service: service,
      environment: ProcessInfo.processInfo.environment["APP_ENV"] ?? "dev",
      destination: ProcessInfo.processInfo.environment["BENCH_DESTINATION"] ?? "control",
      exporterEndpoint: ProcessInfo.processInfo.environment["OTEL_EXPORTER_OTLP_ENDPOINT"]
        ?? "http://collector:4318"
    )
  }

  static func port(fallback: UInt16) -> UInt16 {
    if let raw = ProcessInfo.processInfo.environment["PORT"], let p = UInt16(raw) {
      return p
    }
    return fallback
  }
}
